tests/MongoDB.Driver.Core.Tests/Core/Operations/CreateIndexRequestTests.cs

```diff
@@ -598,6 +599,47 @@ public void CreateIndexDocument_should_return_expected_result()
             result.Should().Be(expectedResult);
         }
 
+        [Theory]
+        [ParameterAttributeData]
+        public void CreateIndexDocument_should_return_expected_result_when_wildcard_index_is_set(
+            [Values("{ '$**' : 1 }", "{ 'tags.$**' : 1 }")] string wildcardIndexString)
+        {
+            var wildcardIndexKey = BsonDocument.Parse(wildcardIndexString);
+            var subject = new CreateIndexRequest(wildcardIndexKey);
+
+            var result = subject.CreateIndexDocument(null);
+
+            var expectedResult = new BsonDocument
+            {
+                { "key", wildcardIndexKey },
+                { "name", wildcardIndexKey.GetElement(0).Name + "_1" }
+            };
+            result.Should().Be(expectedResult);
+        }
+
+        [Theory]
+        [ParameterAttributeData]
+        public void CreateIndexDocument_should_return_expected_result_when_wildcardProjection_is_set(
+            [Values(null, "{ x : 123 }", "{ x : 123, y : 234 }")] string wildcardProjectionString)
+        {
+            var keys = new BsonDocument("x.$**", 1);
+            var wildcardProjection = wildcardProjectionString == null ? null : BsonDocument.Parse(wildcardProjectionString);
+            var subject = new CreateIndexRequest(keys)
+            {
+                WildcardProjection = wildcardProjection
+            };
+
+            var result = subject.CreateIndexDocument(null);
+
+            var expectedResult = new BsonDocument
+            {
+                { "key", keys },
+                { "name", "x_1" },
```

Comment:
I had to change this line to get the tests to pass:

    { "name", "x.$**_1" }

Reply:
Yeah, I broke it with the last commit :( 
Done